src/cms/page_xliff_converter.py

```diff
@@ -362,76 +278,132 @@ def _get_page_translation_slug(title):
+        file_paths = []
+        with ZipFile(zip_file_path, 'r') as zip_ref:
+            zip_ref.extractall(os.path.dirname(zip_file_path))
+            for file_name in zip_ref.namelist():
+                file_paths.append(os.path.join(os.path.dirname(zip_file_path), file_name))
+        return file_paths
+
+    def generate_xliff_import_diff(self, xliff_paths):
+        """
+        Generate diff between XLIFF content and current translation content
+
+        :param xliff_paths: list of paths to XLIFF files
+        :type xliff_paths: [ str ]
+        :return: dictionaries containing diffs between XLIFF and current translation versions
+        :rtype: [ dict ]
+        """
+        diffs = []
+        for xliff_path in xliff_paths:
+            if xliff_path.endswith(('.xlf', '.xliff')) and os.path.isfile(xliff_path):
+                with open(xliff_path, 'r', encoding='utf-8') as f:
+                    xliff_content = f.read()
+                    converter = TranslationXliffConverter(xliff_code=xliff_content)
+                    trans_fields = converter.xliff_to_translation_data()
+                    if trans_fields is None:
+                        continue
+                    diffs.append(self.generate_translation_diff(trans_fields, os.path.basename(xliff_path)))
+        return diffs
 
-    def import_xliffs_zip_file(self, zip_file_path, user):
-        results = []
-
-        if zip_file_path.startswith(XLIFFS_DIR) and zip_file_path.endswith('.zip') and os.path.isfile(zip_file_path):
-            with ZipFile(zip_file_path, 'r') as zip_file:
-                for file_name in zip_file.namelist():
-                    if file_name.endswith(('.xliff', '.xlf',)):
-                        with zip_file.open(file_name) as f:
-                            try:
-                                xliff_content = f.read()
-                                page_xliff = self.converter.xliff_to_page_xliff(xliff_content)
-                                if page_xliff:
-                                    results.append((file_name, self.save_page_xliff(page_xliff, user),))
-                                else:
-                                    results.append((file_name, False,))
-                            except XliffValidationException:
-                                results.append((file_name, False,))
-        return results
+    @staticmethod
+    def generate_translation_diff(trans_fields, xliff_name):
+        """
+        Generate diff to current translation from translation fields dictionary
+
+        :param trans_fields: content extracted from XLIFF
+        :type trans_fields: dict
+        :param xliff_name: file name of XLIFF
+        :type xliff_name: str
+        :return: dictionary containing diff between XLIFF and current translation version
+        :rtype: dict
+        """
+        page = Page.objects.get(id=int(trans_fields["page_id"]))
+        tgt_trans = page.get_translation(trans_fields["tgt_lang_code"])
```

Comment:
Also, I get an error when I want to import  a translation which does not yet exist... so I suggest to use [get_or_create()](https://docs.djangoproject.com/en/2.2/ref/models/querysets/#get-or-create) here as well.